products/cloudasset/api.yaml

```diff
@@ -0,0 +1,252 @@
+      - !ruby/object:Api::Type::String
+        name: feedId
+        description: |
+          This is the client-assigned asset feed identifier and it needs to be unique under a specific parent.
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::Array
+        name: assetNames
+        item_type: Api::Type::String
+        description: |
+          A list of the full names of the assets to receive updates. You must specify either or both of 
+          assetNames and assetTypes. Only asset updates matching specified assetNames and assetTypes are
+          exported to the feed. For example: //compute.googleapis.com/projects/my_project_123/zones/zone1/instances/instance1.
+          See https://cloud.google.com/apis/design/resourceNames#fullResourceName for more info.
+      - !ruby/object:Api::Type::Array
+        name: assetTypes
+        item_type: Api::Type::String
+        description: |
+          A list of types of the assets to receive updates. You must specify either or both of assetNames
+          and assetTypes. Only asset updates matching specified assetNames and assetTypes are exported to
+          the feed. For example: "compute.googleapis.com/Disk"
+          See https://cloud.google.com/asset-inventory/docs/supported-asset-types for a list of all
+          supported asset types.
+      - !ruby/object:Api::Type::String
+        name: contentType
+        description: |
+          Asset content type. If not specified, no content but the asset name and type will be returned.
+      - !ruby/object:Api::Type::NestedObject
+        name: feedOutputConfig
+        required: true
+        description: |
+          Output configuration for asset feed destination.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: pubsubDestination
+            description: |
+              Destination on Cloud Pubsub.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: topic
+                description: |
+                  Destination on Cloud Pubsub topic.
+  - !ruby/object:Api::Resource
+    name: FolderFeed
+    base_url: "{{folder}}/feeds"
+    create_url: "{{folder}}/feeds?feedId={{feed_id}}"
+    self_link: "{{name}}"
+    update_verb: :PATCH
+    update_mask: true
+    collection_url_key: 'feeds'
+    description: |
+      Describes a Cloud Asset Inventory feed used to to listen to asset updates.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/asset-inventory/docs'
+      api: 'https://cloud.google.com/asset-inventory/docs/reference/rest/'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: folder
+        required: true
+        input: true
+        url_param_only: true
+        description: |
+          The folder this feed should be created in.
+      - !ruby/object:Api::Type::String
```

Comment:
Actually, I need this. This is where this API is a bit different. When creating a feed, you need to provide the "billing project" that will be used for the feed in the X-Goog-User-Project header. But this project id will not appear in the resource URL.
I enabled the supports_indirect_user_project_override flag because I needed the project variable to be defined and passed to the sendRequestWithTimeout function. Otherwise a hard coded empty string is passed.
This is where I am not happy in the way we are passing this billing project today. The current implementation of the sendRequestWithTimeout function has been designed only for a configuration global billing project. But this API requires the flexibility to use different billing projects for different feeds.
The billing project affects the following:
1.- It is the project that will need to have the cloudasset.googleapis.com service enabled.
2.- It is the project whose cloudasset identity will be used for publishing messages to the destination topic.
So we need full flexibility here. And this is why in the projectFeed resource we have a "project" parameter and a "billing_project". If no billing_project is provided, the "project" will also be used as billing_project.